flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -441,8 +435,26 @@ private int getRestPortFromExternalService(Service externalService) {
                 address = loadBalancer.getIngress().get(0).getHostname();
             }
         } else {
-            // Use node port
-            address = this.internalClient.getMasterUrl().getHost();
+            // Use node port. Node port is accessible on any node within kubernetes cluster. We'll
+            // only consider IPs with the configured address type.
+            address =
+                    internalClient.nodes().list().getItems().stream()
+                            .flatMap(node -> node.getStatus().getAddresses().stream())
+                            .filter(
+                                    nodeAddress ->
+                                            nodePortAddressType
+                                                    .name()
+                                                    .equals(nodeAddress.getType()))
+                            .map(NodeAddress::getAddress)
+                            .filter(externalIp -> !externalIp.isEmpty())
```

Comment:
Small nit: Would it make more sense to name this `address`, rather than `externalIp`, since I assume the map above return an address? As I understand, the address we grab here might be external or internal depending on what the expoed node port address type is set to?

Reply:
address is already defined in the scope, I'll change this to `ip`